model/src/cfd.rs

```diff
@@ -1336,6 +1333,41 @@ impl Cfd {
         Ok((tx, sig, script_pk))
     }
 
+    /// Number of hours that the time-to-live of the contract will be
+    /// extended by with the next rollover.
+    ///
+    /// During rollover the time-to-live of the contract is extended
+    /// so that the non-collaborative settlement time is set to ~24
+    /// hours in the future from now.
+    fn hours_to_extend_in_rollover(&self) -> Result<u64> {
+        let dlc = self.dlc.as_ref().context("Cannot roll over without DLC")?;
+        let settlement_time = dlc.settlement_event_id.timestamp();
+
+        let hours_left = settlement_time - OffsetDateTime::now_utc();
+
+        if !hours_left.is_positive() {
+            tracing::warn!("Rolling over a contract that can be settled non-collaboratively");
+
+            return Ok(SETTLEMENT_INTERVAL.whole_hours() as u64);
+        }
+
+        let time_to_extend = SETTLEMENT_INTERVAL
+            .checked_sub(hours_left)
+            .with_context(|| {
+                format!(
+                    "Cannot rollover if time-to-live of contract is < {} hours",
+                    SETTLEMENT_INTERVAL.whole_hours()
+                )
+            })?;
+        let hours_to_extend = time_to_extend.whole_hours();
+
+        if !hours_to_extend.is_positive() {
+            bail!("Rollover must extend time-to-live by at least 1 hour");
+        }
```

Comment:
Shouldn't we just default to `1` here?

Do we have a test that hits this path?

Reply:
I thought it would be undesirable to rollover if we already have a big enough time-to-live.

To make it less prone to failures I've made it so that a CFD with a time-to-live of 24-25 hours actually rolls over for an extra hour. Beyond that I think it's okay to prevent the rollover from happening. Let me know what you think about https://github.com/itchysats/itchysats/pull/1618/commits/3a672f26d090e45eb39a68dbc16b19371e4af075. I've also added a test.

Reply:
Yes sounds good to me!